File: tests/sdk/State.ts
import { PublicKey } from "@solana/web3.js";
import { BN } from "@project-serum/anchor";

//@formatted:off
export interface State {
  admin: PublicKey          // 32
  stateBump: number,        // 1
  signerBump: number,       // 1
  counter: BN,              // 2
}

File: programs/anchor-example/src/structs/state.rs
use anchor_lang::prelude::*;

#[account]
#[derive(PartialEq, Default, Debug)]
//@formatter:off
pub struct State {                          // 8 - account signature
    pub admin: Pubkey,                      // 32
    pub state_bump: u8,                     // 1
    pub signer_bump: u8,                    // 1
    pub counter: u16,                       // 2
}                                           // XX - total
//@formatter:on
